### PR TITLE
RELATED: RAIL-3966 - AttributeFilterButton examples fixes

### PR DESCRIPTION
**Pull request by plavemar**
- compare previous and actual parent filters after placeholder resolution
- handling components depending on child filter in the parent component
- attributeFilterButton always defined by refs in examples
- conidering title prop in AttributeFilterButton subtitle resolution

There is a need to handle child-filter-dependent components in all AttributeFilterButton parent-child examples, hence the effects reflecting the parent-filter changes.

In `getItemsTitles` function in `AttributeFilterUtils` the support for using attribute element title property if the filter is defined by values instead of refs. In the same function, we accept the possibility that the element's title will be resolved as undefined, that's why resolved titles are in addition filtered to remove falsy values.

JIRA: RAIL-3966, RAIL-3956

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6348/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8155/ : SUCCESS in 17m 19s

File: examples/sdk-examples/src/examples/attributeFilter/AttributeParentChildFilterButtonWithPlaceholderExample.tsx [hunk 1]
```diff
@@ -52,6 +53,19 @@ const AttributeParentChildFilterButtonWithPlaceholder: React.FC = () => {
         console.info("AttributeFilterExample onLoadingChanged", ...params);
     };
 
+    const parentFilter = useResolveValueWithPlaceholders(stateFilterPlaceholder);
+
+    /**
+     * What happens with the component depending on the child filters is handled outside the AttributeFilterButton component.
+     */
+    useEffect(() => {
+        cityFilterPlaceholder = newPlaceholder<IAttributeFilter>(
```
Comment (no23reason) [inline examples/sdk-examples/src/examples/attributeFilter/AttributeParentChildFilterButtonWithPlaceholderExample.tsx:62]: This looks sketchy, should this be in state?

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2094

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1839/ : SUCCESS in 59s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2120/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/921/ : SUCCESS in 14m 22s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8161/ : SUCCESS in 20m 24s

**Pull request merged**